examples/postgres/advanced-blog-cli/src/auth.rs

```diff
@@ -0,0 +1,164 @@
+use bcrypt::*;
+use diesel::prelude::*;
+use diesel::{self, insert_into};
+use dotenv;
+
+use schema::users;
+
+#[derive(Debug)]
+pub enum AuthenticationError {
```

Comment:
I'd move this (and the From impl) just above the helper functions (that start with `find_user`).